zerver/models/saved_replies.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any
+
+from django.db import models
+from django.utils.timezone import now as timezone_now
+
+from zerver.models.users import UserProfile
+
+
+class SavedReply(models.Model):
+    MAX_TITLE_LENGTH = 30
+
+    user_profile = models.ForeignKey(UserProfile, on_delete=models.CASCADE)
+    title = models.CharField(max_length=MAX_TITLE_LENGTH)
+    content = models.TextField()
+    rendered_content = models.TextField()
+    date_created = models.DateTimeField(default=timezone_now)
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "id": self.id,
+            "title": self.title,
+            "content": self.content,
+            "rendered_content": self.rendered_content,
+            "date_created": int(self.date_created.timestamp()),
+        }
```

Comment:
I was thinking about whether we need any indexes, but probably the `user_profile` `ForeignKey` is the only thing that would end up being used. But maybe add a `realm` column? We've denormalized that immutable field into a decent number of other tables, and we'll need it if in the future we add shared saved replies.

Reply:
Added `realm` field.